specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/sqlX.json

```diff
@@ -0,0 +1,257 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Cosmos DB",
+    "description": "Azure Cosmos DB Database Service Resource Provider REST API",
+    "version": "2020-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "Impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/services/sqlx": {
```

Comment:
This appears to be a singleton resource type. Can the user create more than one databaseAccounts/services resource? If not, this should likely be named "default" and the resource type should reflect that it is sqlx. Otherwise, should we make "sqlx" a parameter like {serviceName}? I am not sure what ARM's guidelines are for fixing a name, but it appears as though we are distinguishing the behavior of the "service" resource based on the name, which differs from how our other ARM resources operate.

Reply:
Just saw your comment about making sqlx a parameter, but I am still wondering about the nature of this resource. Can a user choose the name, and can they have multiple?

Reply:
For now the user cannot choose the name, we have sqlx, janus and MVBuilder computev2 services. User can only choose between these names (once mvbuilder is also available). In future we might have some applications where user will be able to choose the names. Regardless, this is a very good comment, let us discuss this, during the api review meeting.